nexus/db-queries/src/db/pool.rs

```diff
@@ -177,19 +189,14 @@ impl Drop for Pool {
         // it's possible for these tasks to keep nudging slightly forward if
         // we're using a multi-threaded async executor.
         //
-        // With this check, we'll reliably panic (rather than flake) if the pool
-        // is dropped without terminating these worker tasks.
+        // With this check, we'll warn if the pool is dropped without
+        // terminating these worker tasks.
         if !self.terminated.load(std::sync::atomic::Ordering::SeqCst) {
-            // If we're already panicking, don't panic again.
-            // Doing so can ruin test handlers by aborting the process.
-            //
-            // Instead, just drop a message to stderr and carry on.
-            let msg = "Pool dropped without invoking `terminate`";
-            if std::thread::panicking() {
-                eprintln!("{msg}");
-            } else {
-                panic!("{msg}");
-            }
+            error!(
+                self.log,
+                "Pool dropped without invoking `terminate`. qorb background tasks
+                 should be cancelled, but they may briefly still be initializing connections"
+            );
```

Comment:
Hmm, perhaps this ought to also be `eprintln!`'d or something if we are running tests? When running Nexus tests, log messages get stuffed in a file that may or may not be inspected when a test fails depending on the nature of the failure. If we printed it to stderr, though, it would always be visible when a test fails...

Reply:
I think it's probably okay as-is, or even as a warning-level message.  As I understand it, this is not a problem at all -- it's only if consumers care about waiting for these tasks to stop running that this matters.